src/ol_infrastructure/applications/mitxpro/__main__.py

```diff
@@ -0,0 +1,216 @@
+from pulumi import StackReference, export, get_stack
+from pulumi.config import get_config
+from pulumi_aws import ec2
+from pulumi_consul import Node, Service
+
+from ol_infrastructure.components.aws.database import OLAmazonDB, OLMariaDBConfig
+from ol_infrastructure.components.services.vault import (
+    OLVaultDatabaseBackend,
+    OLVaultMariaDatabaseConfig,
+)
+from ol_infrastructure.infrastructure.operations.consul import (
+    consul_server_security_group,
```

Comment:
Importing the security group definition won't work for using as a reference. We need to pass the ID of the resource generated, so we'll need to rely on stack outputs from the operations stack.

Reply:
I call the secgroup id [here](https://github.com/mitodl/ol-infrastructure/blob/4aa2c65b3cdee4fcf1c95a6c477a3bf29eb756ae/src/ol_infrastructure/applications/mitxpro/__main__.py#L157). Wouldn't that work?

Reply:
No, because that will cause it to try to execute that whole module and then return the output of the `id` object. Despite it being "just Python" we can't import between projects like that. The only interface for data exchange across concrete projects is via the stack references.

Reply:
Essentially, by importing from the Consul module it will try to deploy all of the resources defined within it.

Reply:
Isn't [this](https://github.com/mitodl/ol-infrastructure/blob/95aca39329813d610dbfd1efc7e51c4dd06105c1/src/ol_infrastructure/applications/edx/__main__.py#L57) though basically doing the same?

Reply:
The difference is that for that import it's within the same project and it is intended to deploy all of those resources. If we import it from a different project then it will cause all of those resources to be deployed again but with the wrong stack information.

Reply:
Makes sense, but the stack that we're trying to get the secgroup.id from (consul in this case) has to be exporting the `consul_server_security_group`, right?

Reply:
Correct, which it already does. https://github.com/mitodl/ol-infrastructure/blob/main/src/ol_infrastructure/infrastructure/operations/__main__.py#L16

Reply:
https://github.com/mitodl/ol-infrastructure/blob/main/src/ol_infrastructure/infrastructure/operations/consul.py#L166-L169